ProjectName/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  NetworkAPIProtocol.swift
+//
+
+import RxSwift
+import Alamofire
+import RxAlamofire
+
+protocol NetworkAPIProtocol {
+
+    func performRequest<T: Decodable>(_ configuration: RequestConfiguration, for type: T.Type) -> Single<T>
+}
+
+extension NetworkAPIProtocol {
+
+    func request<T: Decodable>(
+        session: Session,
+        configuration: RequestConfiguration,
+        decoder: JSONDecoder
+    ) -> Single<T> {
+        return session.rx.request(
+            configuration.method,
+            configuration.url,
+            parameters: configuration.parameters,
+            encoding: configuration.encoding,
+            headers: configuration.headers,
+            interceptor: configuration.interceptor
+        )
+        .responseData()
+        .observe(on: ConcurrentDispatchQueueScheduler(qos: .background))
```

Comment:
I think we can remove this line also, we will control about `observer(on:)` in use case is better